### ISSUE TITLE
Tokenization Incorrect

### ISSUE THREAD

**Issue by dhirajmadan1** [huggingface/pytorch-pretrained-BERT#266]
The tokenizer is not working correctly for me for e.g. [CLS] is gettig broken '[' , 'cl ', '##s', ']'
In [1]: import torch 
   ...: from pytorch_pretrained_bert import BertTokenizer, BertModel, BertForMas
   ...: kedLM 
   ...:  
   ...: # Load pre-trained model tokenizer (vocabulary) 
   ...: tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')          
Better speed can be achieved with apex installed from https://www.github.com/nvidia/apex.

In [2]: text = "[CLS] Who was Jim Henson ? [SEP] Jim Henson was a puppeteer [SEP
   ...: ]" 
   ...: tokenized_text = tokenizer.tokenize(text)                               

In [3]: tokenized_text                                                          
Out[3]: 
['[',
 'cl',
 '##s',
 ']',
 'who',
 'was',
 'jim',
 'henson',
 '?',
 '[',
 'sep',
 ']',
 'jim',
 'henson',
 'was',
 'a',
 'puppet',
 '##eer',
 '[',
 'sep',
 ']']

**Comment by wangxiaodiu** [comment]
Me too!

I followed the instruction from the readme, the `tokenized_text` is expected by assertion to be:
 `['[CLS]', 'who', 'was', 'jim', 'henson', '?', '[SEP]', 'jim', '[MASK]', 'was', 'a', 'puppet', '##eer', '[SEP]']`. 

However, the actual `tokenized_text` is:
`['[CLS]', 'who', 'was', 'jim', 'henson', '?', '[MASK]', 'jim', 'henson', 'was', 'a', 'puppet', '##eer', '[SEP]']`.

I believe this is because `masked_index = 6` in the example code from readme. If you let it to `masked_index = 8`, everything is perfect.

**Comment by bsobolik** [comment]
I think the tokenizer issue has been resolved in the latest version (0.5.0).

**Comment by thomwolf** [comment]
Yes!

**Comment by yenicelik** [comment]
I just encountered the same issue as @dhirajmadan1 with `transformers==2.2.1`. Is this expected somehow?

I am following the quickstart guide: https://huggingface.co/transformers/quickstart.html

```
tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')

        # Run an example text through this:
        text = "[CLS] Who was Jim Henson ? [SEP] Jim Henson was a puppeteer [SEP]"
        tokenized_text = tokenizer.tokenize(text)

        masked_index = 8
        tokenized_text[masked_index] = '[MASK]'
        predicted_tokenized_sentence = ['[CLS]', 'who', 'was', 'jim', 'henson', '?', '[SEP]', 'jim', '[MASK]', 'was', 'a',
                                  'puppet', '##eer', '[SEP]']
        
```

**Comment by LysandreJik** [comment]
What is your issue @yenicelik? Do you mind opening a new issue with your problem?

**Comment by yenicelik** [comment]
ah, my apologies: https://github.com/huggingface/transformers/issues/2047

apparently a PR is on the way!
